mllib/src/test/scala/org/apache/spark/ml/regression/DecisionTreeRegressorSuite.scala

```diff
@@ -96,6 +108,15 @@ class DecisionTreeRegressorSuite
       assert(variance === expectedVariance,
         s"Expected variance $expectedVariance but got $variance.")
     }
+
+    val toyDF = TreeTests.setMetadata(toyData, Map.empty[Int, Int], 0)
+    dt.setMaxDepth(1)
+      .setMaxBins(6)
+      .setSeed(0)
+    val expectVariances = dt.fit(toyDF).transform(toyDF).select("variance").collect().map {
```

Comment:
`expectedVariances` and `trueVariances` are mixed up here. Expected should be the theoretical value computed below. Also, it would be good to leave a comment explaining where those expected values came from.